sdk/core/azure-core/src/test/java/com/azure/core/http/rest/PagedFluxTest.java

```diff
@@ -284,15 +284,63 @@ public void testPagedFluxWithContext() throws Exception {
             return Mono.empty();
         }
 
-        return Mono.just(pagedResponses.get(Integer.valueOf(continuationToken)));
+        int parsedToken = Integer.parseInt(continuationToken);
+        if (parsedToken >= pagedResponses.size()) {
+            return Mono.empty();
+        }
+
+        return Mono.just(pagedResponses.get(parsedToken));
     }
 
     private List<Integer> getItems(Integer i) {
         return IntStream.range(i * 3, i * 3 + 3).boxed().collect(Collectors.toList());
     }
 
     private List<String> getStringItems(Integer i) {
-        return IntStream.range(i * 3, i * 3 + 3).boxed().map(val -> String.valueOf(val)).collect(Collectors.toList());
+        return IntStream.range(i * 3, i * 3 + 3).boxed().map(String::valueOf).collect(Collectors.toList());
     }
 
+    @Test
+    public void streamFindFirstOnlyRetrievesOnePage() throws InterruptedException {
+        OnlyOnePageRetriever pageRetriever = new OnlyOnePageRetriever();
+        Integer next = new OnlyOnePagedFlux(() -> pageRetriever).blockFirst();
+
+        Thread.sleep(2000);
+
+        assertEquals(1, pageRetriever.getGetCount());
+    }
```

Comment:
I would add some guard/assertion to all these new tests that makes sure that there's actually more than one page in the underlying data source. Otherwise it's easy to make this test worthless if someone changes underlying data source to have just one page. (btw. it's not that easy for me to trace data flow in these tests so I can't tell whether source has one or more pages... some check would be great.)

Reply:
`OnlyOnePageRetriever` should return 5 pages when consumed to completion. In each of the new tests I'll add an initial verification that `OnlyOnePagedFlux`/`OnlyOnePagedIterable` using collect `toList` is greater than one element.